symfony/plugins/orangehrmAdminPlugin/Api/LicenseAPI.php

```diff
@@ -0,0 +1,165 @@
+
+    /**
+     * @throws Exception
+     * @return LicenseService
+     */
+    public function getLicenseService(): LicenseService
+    {
+        if (is_null($this->licenseService)) {
+            $this->licenseService = new LicenseService();
+        }
+        return $this->licenseService;
+    }
+
+    /**
+     * @param LicenseService $licenseService
+     */
+    public function setLicenseService(LicenseService $licenseService): void
+    {
+        $this->licenseService = $licenseService;
+    }
+
+    /**
+     * @return EndpointGetOneResult
+     * @throws RecordNotFoundException
+     * @throws DaoException
+     * @throws Exception
+     */
+    public function getOne(): EndpointGetOneResult
+    {
+        // TODO:: Check data group permission
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, self::PARAMETER_ID);
+        $license = $this->getLicenseService()->getLicenseById($id);
+        if (!$license instanceof License) {
+            throw new RecordNotFoundException('No Record Found');
+        }
+
+        return new EndpointGetOneResult(LicenseModel::class, $license);
+    }
+
+    /**
+     * @return EndpointGetAllResult
+     * @throws DaoException
+     * @throws Exception
+     */
+    public function getAll(): EndpointGetAllResult
+    {
+        // TODO:: Check data group permission
+
+        $licenseParamHolder = new LicenseSearchFilterParams();
+        $this->setSortingAndPaginationParams($licenseParamHolder);
+        $licenses = $this->getLicenseService()->getLicenseList($licenseParamHolder);
+        $count = $this->getLicenseService()->getLicenseCount($licenseParamHolder);
+        return new EndpointGetAllResult(LicenseModel::class, $licenses, new ParameterBag(['total' => $count]));
+    }
+
+    /**
+     * @inheritDoc
+     * @return EndpointCreateResult
+     * @throws Exception
+     */
+    public function create(): EndpointCreateResult
+    {
+        // TODO:: Check data group permission
+        $licenses = $this->saveLicense();
+
+        return new EndpointCreateResult(LicenseModel::class, $licenses);
+    }
+
+    /**
+     * @inheritDoc
+     * @return EndpointUpdateResult
+     * @throws Exception
+     */
+    public function update(): EndpointUpdateResult
+    {
+        // TODO:: Check data group permission
+        $licenses = $this->saveLicense();
+
+        return new EndpointUpdateResult(LicenseModel::class, $licenses);
+    }
+
+    /**
+     * @return License
+     * @throws RecordNotFoundException
+     * @throws \OrangeHRM\Core\Exception\DaoException
+     */
+    public function saveLicense(): License
+    {
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, self::PARAMETER_ID);
+        $name = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        if (!empty($id)) {
+            $education = $this->getLicenseService()->getLicenseById($id);
```

Comment:
Is $education variable name correct?

Reply:
@KaveeshaSilva1996 no actually It was from the education API i created